pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2594,7 +2596,23 @@ func parseEnvVariablesFromCloudInit(envPairs []string) (map[string]string, error
 			errStr := fmt.Sprintf("Variable \"%s\" not defined properly\nKey value pair should be delimited by \"=\"", pair[0])
 			return nil, errors.New(errStr)
 		}
-		envList[pair[0]] = pair[1]
+
+		// Trim off (i.e., remove leading and trailing) spaces from
+		// key/value pair so the export command can be easily generated in
+		// the KEY=VALUE format
+		key := strings.Trim(pair[0], " ")
+		val := strings.Trim(pair[1], " ")
+
+		// Parse the variable declaration to search for any syntax error
+		// that would break the processing shell
+		expline := key + "=" + val
+		_, err := envp.Parse(bytes.NewReader([]byte(expline)))
+		if err != nil {
+			errStr := fmt.Sprintf("Variable \"%s\" not defined properly: %s", key, err)
+			return nil, errors.New(errStr)
```

Comment:
no need to first create a string and then convert it into an `
error`
```suggestion
			return nil, fmt.Errorf("Variable \"%s\" not defined properly: %s", key, err)
```